src/Normalizer/NodeNormalizer.php

```diff
@@ -36,7 +37,55 @@ public function normalize($entity, $format = NULL, array $context = []) {
       $normalized['is_restricted'] = ['value' => $lessonHandler->isRestricted($entity)];
     }
 
+    /** @var \Drupal\anu_lms\Course $courseHandler */
+    $courseHandler = \Drupal::service('anu_lms.course');
+
+    if ($entity->bundle() === 'course' && $courseHandler->isLinearProgressEnabled($entity)) {
+      $normalized['progress'] = ['value' => $this->getCourseProgress($entity)];
+    }
+
     return $normalized;
   }
 
+  /**
+   * Returns progress in a course.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $course
+   *   Course object.
+   *
+   * @return array
+   *   The passed array with the progress attached.
+   */
+  protected function getCourseProgress(EntityInterface $course) {
+    /** @var \Drupal\anu_lms\Lesson $lessonHandler */
+    $lessonHandler = \Drupal::service('anu_lms.lesson');
+
+    $modules = $course->get('field_course_module')->referencedEntities();
+    $totalLessons = 0;
+    $completedLessons = 0;
+    foreach ($modules as $module) {
```

Comment:
@rodrigoaguilera I wonder if we should consider quizzes as well for the progress tracking. What do you think about that?

Reply:
My understanding is that a quiz at the end of a course is optional and that can complicate the logic of completing the last step of a course but since is simple enough by just adding the quiz to the percentage count if exists I can add it to this PR.

Reply:
I'm not sure how to proceed either - let's check with the team and see what they say, especially curious to hear Elise's opinion on this.